src/engines/fx/lpc/lpc.cpp

```diff
@@ -21,69 +24,73 @@ namespace otto::engines {
 
   LPC::LPC() : EffectEngine("LPC", props, std::make_unique<LPCScreen>(this)) {}
 
-  void LPC::makeSha(gsl::span<float> buffer, float sha_period, int lag)
+  void LPC::makeSha(gsl::span<float> buffer, int sha_period)
   {
     util::fill(buffer, 0.f);
-    for (int i = lag; i < buffer.size(); i += sha_period) {
+    for (int i = this->lag; i < buffer.size(); i += sha_period) {
       buffer[i] = 1;
     }
-    lag = sha_period - buffer.size()%sha_period;
+    this->lag = sha_period - (int)(buffer.size()%sha_period);
   }
 
-  audio::ProcessData<2> LPC::process(audio::ProcessData<2> data)
+  audio::ProcessData<2> LPC::process(audio::ProcessData<1> data)
   {
-    // TODO: Add processing
-    #define ORDER 25
-    for(int i=0;i<ORDER;++i) acov[i] = data[i];
-    acov = acovb(acov);
+    auto in = data.audio;
+    int N = (int) in.size();
+    for(int i=0;i<N;++i) acov[i] = in[i];
+    acovb(acov);
 
     // Vocal signals usually have fundamentals in the [20Hz; 600Hz] band
     const float f_min = 20;
     const float f_max = 600;
 
     const float sampling_freq = 44100;
 
-    const maxT = ceil(sampling_freq/f_min);
-    const minT = floor(sampling_freq/f_max);
+    const int maxT = (int) ceil(sampling_freq/f_min);
+    const int minT = (int) floor(sampling_freq/f_max);
 
-    auto pitch = detect_pitch(gsl::span<float> data, unsigned minT, unsigned maxT);
+    auto pitch = detect_pitch(gsl::span(in.begin(), in.size()), minT, maxT);
```

Comment:
`in` should be automatically convertible to a span. If not, do `gsl::span(in.begin(), in.end())`